articles/articles.go

```diff
@@ -0,0 +1,77 @@
+package articles
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	dperrors "github.com/ONSdigital/dp-api-clients-go/v2/errors"
+	"github.com/ONSdigital/dp-api-clients-go/v2/headers"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// GetLegacyBulletin returns a legacy bulletin
+func (c *Client) GetLegacyBulletin(ctx context.Context, userAccessToken, collectionID, lang, uri string) (*Bulletin, error) {
+	url := fmt.Sprintf("%s/articles/legacy?url=%s&lang=%s", c.hcCli.URL, uri, lang)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, dperrors.New(
+			fmt.Errorf("failed to create request to Articles API: %s", err),
+			http.StatusInternalServerError,
+			nil,
+		)
+	}
+
+	headers.SetCollectionID(req, collectionID)
+	headers.SetAuthToken(req, userAccessToken)
+
+	resp, err := c.hcCli.Client.Do(ctx, req)
+	if err != nil {
+		return nil, dperrors.New(
+			fmt.Errorf("failed to get response from Articles API: %s", err),
+			http.StatusInternalServerError,
+			nil,
+		)
+	}
+	defer closeResponseBody(ctx, resp)
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, c.errorResponse(resp)
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, dperrors.New(
+			fmt.Errorf("failed to read response body from Articles API: %s", err),
+			resp.StatusCode,
+			nil,
+		)
+	}
+
+	if len(b) == 0 {
```

Comment:
Is this not a bit odd - it looks like we are goign to force a json.Unmarshal error? Why not just return an error taht describes what happened, like:
dperrors.New(
			"empty response"
			http.StatusInternalServerError,
			nil},
		)

Reply:
It does look odd indeed and I asked myself the same question. 
I just followed the pattern in this library as used in other sections but I think the idea of this is to explicitly output that it came empty rather than displaying an empty string in the error. The unmarshaling will fail anyway when `b` is empty.

We could remove it if we feel it is pointless. Otherwise, I think I should add a test case for it